src/anomalib/metrics/per_image/binclf_curve.py

```diff
@@ -0,0 +1,178 @@
+"""Binary classification curve (torch interface).
+
+This module implements torch interfaces to access the numpy code in `binclf_curve_numpy.py`.
+Check its docstring for more details.
+
+Tensors are build with `torch.from_numpy` and so the returned tensors will share the same memory as the numpy arrays.
+
+Validations will preferably happen in ndarray so the numpy code can be reused without torch,
+so often times the Tensor arguments will be converted to ndarray and then validated.
+
+author: jpcbertoldo
+"""
+
+# Original Code
+# Copyright (c) 2024 @jpcbertoldo
+# https://github.com/jpcbertoldo/aupimo
+# SPDX-License-Identifier: MIT
+#
+# Modified
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
```

Comment:
@ashwinvaidya17 
> Can we avoid this? Last I checked `__future__.annotations` does not behave well with jsonargparse

i thought it was necessary for the annotations like `range: tuple[float, float]`, is it not?

Reply:
So, (i think) i re-figured out why i used this. 
In the other file, the class `AUPIMOResult` has annotations using the class itself (eg `from_pimoresult(...) -> AUPIMOResult:`) and apparently the linter didnt like it, but adding `__future__.annotations` solves it.
Now, idk if this is a good justification. What do you prefer to do about this?